pkg/collector/upgrade/versions.go

```diff
@@ -18,12 +18,17 @@ import (
 	"github.com/Masterminds/semver/v3"
 
 	"github.com/open-telemetry/opentelemetry-operator/apis/v1alpha1"
+	"github.com/open-telemetry/opentelemetry-operator/apis/v1beta1"
 )
 
+// Deprecated use upgradeFuncV1beta1
 type upgradeFunc func(u VersionUpgrade, otelcol *v1alpha1.OpenTelemetryCollector) (*v1alpha1.OpenTelemetryCollector, error)
+type upgradeFuncV1beta1 func(u VersionUpgrade, otelcol *v1beta1.OpenTelemetryCollector) (*v1beta1.OpenTelemetryCollector, error)
 
 type otelcolVersion struct {
-	upgrade upgradeFunc
+	// deprecated use upgradeV1beta1.
+	upgrade        upgradeFunc
+	upgradeV1beta1 upgradeFuncV1beta1
```

Comment:
I didn't want to rewrite old upgrade procedures hence I am adding this API and just adopting tests to work (they need a valid collector spec because v1beta1 is a minimal doing validation)

Reply:
i wonder if we should also include some deprecations around these collector version upgrades. like do we need to keep around something that upgrades collector from 0.30.0 to 0.31.0?